### PR TITLE
a few things were done here : there is now a random index choosen and…

### PR DESCRIPTION
**Pull request by QuentinVerwaerde**
… passed as a state that will allow to select a random popup and display inside of it a picture of Anat. I also added a new array made of hints for each paris' district. We then create an array with a random set object-array that will display the hints at random for the corresponding location of Anat. This push passes all the backlog specification

### CODE REVIEW

File: frontend/src/components/Markers.jsx [hunk 1]
```diff
@@ -1,28 +1,18 @@
 import "../style.css";
-import React, { useState, useEffect } from "react";
+import React from "react";
+import PropTypes, { shape } from "prop-types";
 import { MapContainer, TileLayer, Marker, Popup } from "react-leaflet";
 import { Icon } from "leaflet";
 import "leaflet/dist/leaflet.css";
-// import data from "./src/data/data.json";
-import popUimagearray from "../data/data.json";
+// import data from "./src/data/....json";
```
Comment (ioayoub) [inline frontend/src/components/Markers.jsx:7]: 👀

File: frontend/src/components/Markers.jsx [hunk 2]
```diff
@@ -1,28 +1,18 @@
...
-    fetch(
-      "https://opendata.paris.fr/api/explore/v2.1/catalog/datasets/arrondissements/records?limit=20"
-    )
-      .then((res) => res.json())
-      .then((data) => setArrParis(data));
-  }, []);
-
   return (
     arrParis && (
       // création du MapContainer et du TileLayer, indispensable au bon affichage de la carte
```
Comment (ioayoub) [inline frontend/src/components/Markers.jsx:18]: Not really needed here

File: frontend/src/components/hints/Hints.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    pictureAnatorParis.hint3,
+    pictureAnatorParis.hint4,
+    pictureAnatorParis.hint5,
+  ];
+  let randomIndex = "";
+  const arrayHintstoDisplay = [];
+  const gettingRandomNumberupTo5 = (max) => {
+    randomIndex = Math.floor(Math.random() * max);
+  };
+  // merci Ayoub pour l'aide ^^
```
Comment (ioayoub) [inline frontend/src/components/hints/Hints.jsx:20]: <3, but the not really needed here ;)

File: frontend/src/components/hints/Hints.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  const indexNumbers = new Set();
+  while (indexNumbers.size < 3) {
+    gettingRandomNumberupTo5(5);
+    indexNumbers.add(randomIndex);
+  }
+  const newArray = [...indexNumbers];
+  for (const index of newArray) {
+    arrayHintstoDisplay.push(newHintsArray[index]);
+  }
+  // pour l'instant on code en dur les hints mais on pourra dans le futur les rendre plus dynamiques avec des states en fonction du click bouton
```
Comment (ioayoub) [inline frontend/src/components/hints/Hints.jsx:30]: You can remove this comment and add a map to display your hints

File: frontend/src/main.jsx [hunk 5]
```diff
@@ -13,7 +13,8 @@ const router = createBrowserRouter([
 const root = ReactDOM.createRoot(document.getElementById("root"));
 
 root.render(
-  <React.StrictMode>
-    <RouterProvider router={router} />
-  </React.StrictMode>
+  // <React.StrictMode>
```
Comment (ioayoub) [inline frontend/src/main.jsx:16]: My bad, you can set the Strict mode back

File: frontend/src/pages/Home.jsx [hunk 6]
```diff
@@ -1,7 +1,43 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import Markers from "../components/Markers";
+import Hints from "../components/hints/Hints";
+import parisLocationHints from "../data/parisLocationHints.json";
 
 function Home() {
-  return <div>Home</div>;
+  // création de la state pour stocker une api
```
Comment (ioayoub) [inline frontend/src/pages/Home.jsx:7]: Useless comment here

File: frontend/src/pages/Home.jsx [hunk 7]
```diff
@@ -1,7 +1,43 @@
...
+  // on veut géner un comportement aléatoire pour le state, qu'il se mette sur un index au hasard
+  // on créé donc un let aléatoire qui va se coller en index sur le setState
+  const districtIndex = Math.floor(
+    Math.random() * parisLocationHints.length + 1
+  );
+  // création du state pour gérer l'aléatoire dans les pop-ups de Markers et lire son êtat dans hints
+  const [pictureAnatorParis, setPictureAnatorParis] = useState(
+    parisLocationHints[districtIndex]
+  );
+  // ces console.log servent à tricher dans le jeu et est donc indispensable pour la branche dev
```
Comment (ioayoub) [inline frontend/src/pages/Home.jsx:26]: Please make sure to never push dead code (including console.logs), even if needed

**Review by ioayoub — CHANGES_REQUESTED** [review]

File: frontend/src/components/hints/Hints.jsx (hunk 4 above)
Comment (QuentinVerwaerde) [reply frontend/src/components/hints/Hints.jsx:30]: I've removed the comment but I'll refrain from "maping" the array as I'll write some code later (maybe today) using a state and a function onClick to display them while clicking on buttons.

**Review by ioayoub — APPROVED** [review]

**Pull request merged**